README.md

```diff
@@ -426,6 +431,44 @@ fmt.Println(output)
 
 Filters are operations on an existing pipe that also return a pipe, allowing you to chain filters indefinitely.
 
+## Basename
+
+`Basename()` reads a list of filenames from the pipe, one per line, and returns a pipe that contains the last element of each line.
+
+If a line is empty, `Basename()` will return a `.`
```

Comment:
That's the behaviour of the standard Golang library. We could treat it as a special case, and make sure we preserve blank lines.

Reply:
I think we have to do that, because otherwise any input blank lines will be turned into `.`, which isn't a file that exists, so you'd get an error if you tried to open it, for example.